src/DssDirectDepositAaveDai.sol

```diff
@@ -216,25 +224,53 @@ contract DssDirectDepositAaveDai {
 
         emit File(what, data);
     }
+
     function file(bytes32 what, address data) external auth {
+        require(vat.live() == 1, "DssDirectDepositAaveDai/no-file-during-shutdown");
+
         if (what == "king") king = data;
         else revert("DssDirectDepositAaveDai/file-unrecognized-param");
         emit File(what, data);
     }
 
-    // --- Deposit controls ---
-    function wind(uint256 amount) external auth {
-        require(live == 1, "DssDirectDepositAaveDai/not-live");
+    // --- Automated Rate targeting ---
+    function calculateTargetSupply(uint256 targetInterestRate) public view returns (uint256) {
+        require(targetInterestRate > 0, "DssDirectDepositAaveDai/target-interest-zero");
+        require(targetInterestRate <= interestStrategy.getMaxVariableBorrowRate(), "DssDirectDepositAaveDai/above-max-interest");
```

Comment:
These two checks are redundant when the function is called from `exec`. Are they intended for external calls ?

Reply:
Hmmm the `>0` one you are right, but the other is not totally the case. `interestStrategy.getMaxVariableBorrowRate()` could change after you set `bar`.

Reply:
I just checked [the code](https://etherscan.io/address/0xfffE32106A68aA3eD39CcCE673B646423EEaB62a#code), the only way for it to change is if the `interestStategy` changes, which would allow permissionlessly `cage`ing the module.

Reply:
Yeah that's right, still not sure if we should keep it just in case a future strategy does it a bit differently. I'll let @hexonaut to decide on this one.